repos/system_upgrade/el7toel8/actors/checkfstaboptions/actor.py

```diff
@@ -0,0 +1,76 @@
+from collections import defaultdict
+
+from leapp.actors import Actor
+from leapp.exceptions import StopActorExecutionError
+from leapp.models import StorageInfo
+from leapp.reporting import Report
+from leapp.tags import IPUWorkflowTag, ChecksPhaseTag
+from leapp import reporting
+
+
+class CheckFstabOptions(Actor):
+    """
+    Check for mount options that are invalid under RHEL 8.
+
+    Several XFS mount options have been removed from the kernel between RHEL 7 and 8.
+    Their occurence will lead to an unbootable machine. If one of them is detected, the upgrade is inhibited.
+
+    The list of removed options is taken from the xfs(5) manual page on RHEL 8.
+    """
+
+    name = 'check_fstab_options'
+    consumes = (StorageInfo,)
+    produces = (Report,)
+    tags = (IPUWorkflowTag, ChecksPhaseTag)
+
+    REMOVED_XFS_OPTIONS = [
+        # removed from kernel in 4.0
+        'delaylog',
+        'nodelaylog',
+        'ihashsize',
+        'irixsgid',
+        'osyncisdsync',
+        'osyncisosync',
+        # removed from kernel in 4.19
+        'barrier',
+        'nobarrier'
+    ]
+
+    def process(self):
+        storage_info = next(self.consume(StorageInfo), None)
+        if not storage_info:
+            raise StopActorExecutionError(
+                'Could not check fstab entries', details={'details': 'No StorageInfo facts found.'}
+            )
+
+        found = defaultdict(list)
+        for entry in storage_info.fstab:
+            if entry.fs_vfstype == 'xfs':
+                for option in entry.fs_mntops.split(','):
+                    option_name = option.split('=')[0]
+                    if option_name in self.REMOVED_XFS_OPTIONS:
+                        found[entry.fs_spec].append(option_name)
```

Comment:
nothing bad here, just FYI you could simplify the construction from:

for i in iterable:
if condition in terable:
do something

into

for i in filter(or comprhension)(condition, iterable)

in this case you will have a smaller number of lines and more efficient code (looping through generator)

Reply:
Using filter is against our guidelines unless it can't be avoided.

Readable code is preferred over preliminary optimizations

Reply:
@fernflower I am sorry Inna, but I can't see this in our guidelines

Reply:
@ZhukovGreen https://github.com/oamg/leapp-guidelines/blob/master/python-coding-guidelines.md#7-avoid-map-reduce-and-too-complex-comprehensions

Reply:
@fernflower I don't see in my comment any mention of filter or too complex comprehension. Anyway, this is not a strict demand as well as any other from my review

Reply:
Resolving (the code is no longer there).